src/Symfony/Component/Form/Extension/Core/Type/DayOfWeekType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList;
+use Symfony\Component\Form\Extension\Core\DataTransformer\DayOfWeekTransformer;
+
+/**
+ * Choice between all days of the week
+ * Values are <1-7>, 1 = monday, 7 = sunday, ISO-8601
+ * Labels are the localized names (Monday-Sunday)
+ * The first day of the week is governed by the Locale
+ *
+ * @author Sebastien Lavoie <seb@wemakecustom.com>
+ */
+class DayOfWeekType extends AbstractType
+{
+    /**
+     * IntlCalendar::DOW_* constants are not compatible with ISO-8601, redo mapping
+     * @var int[]
+     */
+    private static $intlCalendarMapping = array(
+        \IntlCalendar::DOW_MONDAY    => 1,
+        \IntlCalendar::DOW_TUESDAY   => 2,
+        \IntlCalendar::DOW_WEDNESDAY => 3,
+        \IntlCalendar::DOW_THURSDAY  => 4,
+        \IntlCalendar::DOW_FRIDAY    => 5,
+        \IntlCalendar::DOW_SATURDAY  => 6,
+        \IntlCalendar::DOW_SUNDAY    => 7,
+    );
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $choiceList = function (Options $options) {
+            $calendar = \IntlCalendar::createInstance();
+            $days = DayOfWeekType::getOrderedDaysOfWeek($calendar->getFirstDayOfWeek());
+            $transformer = new DayOfWeekTransformer($options['label_format']);
+            $labels = array_map(array($transformer, 'transform'), $days);
+
+            return new SimpleChoiceList(array_combine($days, $labels));
+        };
+
+        $resolver->setDefaults(array(
+            'label_format' => 'eeee',
+            'choice_list' => $choiceList,
+        ));
+
+        $resolver->setAllowedValues(array(
+            'label_format' => array('e', 'ee', 'eee', 'eeee', 'eeeee', 'eeeeee', 'E', 'EE', 'EEE', 'EEEE', 'EEEEE', 'EEEEEE'),
```

Comment:
Seems like a lot of class constants no ? Simply pointing to the documentation should do the trick. If we are to use class constants, perhaps only add the common ones? Namely:

```php
<?php
const PATTERN_DOW    = 'e';
const PATTERN_SHORT  = 'eee';
const PATTERN_FULL   = 'eeee';
const PATTERN_LETTER = 'eeeee';
```